src/index.js

```diff
@@ -207,22 +219,26 @@ export default function ({Plugin, types: t}) {
             },
 
             JSXOpeningElement(node, parent, scope, file) {
+                const {extractJSX} = getReactIntlOptions(file.opts);
                 const moduleSourceName = getModuleSourceName(file.opts);
 
                 let name = this.get('name');
 
                 if (name.referencesImport(moduleSourceName, 'FormattedPlural')) {
-                    let {loc} = node;
-                    file.log.warn(
-                        `[React Intl] Line ${loc.start.line}: ` +
-                        'Default messages are not extracted from ' +
-                        '<FormattedPlural>, use <FormattedMessage> instead.'
-                    );
+                    const warning = extractJSX ?
+                      createFormattedPluralWarning(node.loc.start.line) :
```

Comment:
This will warn on EVERY visited `FormattedPlural` node which is probably not a great developer experience.